File: 백준/Silver/25644. 최대 상승/최대 상승.java
import java.io.*;
import java.sql.Array;
import java.util.*;



public class Main {






    public static void main(String[] args) throws NumberFormatException, IOException {

        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        BufferedWriter bw = new BufferedWriter(new OutputStreamWriter(System.out));



        int n = Integer.parseInt(br.readLine());

        StringTokenizer st = new StringTokenizer(br.readLine());

        ArrayList<Integer> arr = new ArrayList<Integer>();

        while(st.hasMoreTokens()){
            int a = Integer.parseInt(st.nextToken());
            arr.add(a);
        }


        if(n == 1){
            System.out.println(0);
            return;
        }

        int now  = arr.get(0);
        int ans = 0;
        for(int i = 0 ; i < n ; i++){

            if(now > arr.get(i)){
                now = arr.get(i);
            }else if ( now < arr.get(i)){
                ans = Math.max(ans,arr.get(i)-now);
            }

        }

        System.out.println(ans);









    }








}

File: 백준/Silver/25644. 최대 상승/최대 상승.cc
#include <bits/stdc++.h>
using namespace std;

int main() {

    ios_base::sync_with_stdio(false);
    cout.tie(0);
    cin.tie(0);



    int n;
    cin >> n;

    vector<int>v;

    for(int i = 0 ; i < n ; i ++){
        int a;
        cin >> a;
        v.push_back(a);
    }


    if(n==1){
        cout << 0 << "\n";
    }else{

        int ans = 0;
        int now = v[0];
        for(int i = 1; i < n; i++){

            if(now > v[i]){
                now = v[i];
            }else if (now < v[i]){
                ans = max(ans,v[i]-now);
            }
        }
        cout << ans << "\n";

    }









}

